### PR TITLE
New package: ouch-0.3.1

### PR DESCRIPTION
**Pull request by ElDifinitivo**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

### CODE REVIEW

File: srcpkgs/ouch/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'ouch'
+pkgname=ouch
+version=0.3.1
+revision=1
+build_style=cargo
+hostmakedepends="liblzma-devel bzip2-devel libzip-devel"
```
Comment (abenson) [inline srcpkgs/ouch/template:6]: This should be `makedepends`

File: srcpkgs/ouch/template [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+short_desc="CLI utility for easily compressing and decompressing files and dirs"
+maintainer="Savoy <savoy@liberation.red>"
+license="MIT"
+homepage="https://github.com/ouch-org/ouch"
+distfiles="https://github.com/ouch-org/ouch/archive/refs/tags/${version}.tar.gz"
+checksum=269abaf5ac2f80da3796dbf5e73419c1b64104d1295f3ff57965141f079e6f6d
+
+post_install() {
+	vlicense LICENSE license.txt
```
Comment (abenson) [inline srcpkgs/ouch/template:15]: Just `vlicense LICENSE`, don't rename it

**Comment by ElDifinitivo** [comment]
Forgot to mention but template issues have been resolved

**Review by jcgruenhage — CHANGES_REQUESTED** [review]
`ouch` can generate completions at compile time, which you can enable using

```bash
pre_build() {
	export GEN_COMPLETIONS=1
}
```

then you can do something like this:
```
vcompletion "path/to/_ouch" zsh
vcompletion "path/to/ouch.fish" fish
vcompletion "path/to/ouch.bash" bash
```
in your `post_install` function. 

Aside of completions missing: looks good to me, and looking forward to using `ouch`

**Comment by jcgruenhage** [comment]
So, I was nerd-sniped a bit by this, and wanted to get the completions working at least locally. This turned out to be a slight bit more difficult than anticipated, resulting in two patches I'll submit upstream. One for allowing to override the location of the completions, and one for working around https://github.com/clap-rs/clap/issues/1596.

The resulting template and patches can be found at https://github.com/void-linux/void-packages/commit/44cefdee00b3b8ac0a309819fd3fbfce5e24bb08.

@ElDifinitivo Which one would you prefer, including those changes into your PR, or me opening a PR myself?

**Comment by jcgruenhage** [comment]
Sounds good to me :)

**Comment by paper42** [comment]
@ElDifinitivo please rebase on top of master to fix CI

**Pull request merged**